modules/storage-filesystem/snapshot.go

```diff
@@ -42,54 +42,71 @@ func (m *StorageFileSystemModule) StoreSnapshot(ctx context.Context, snapshot *s
 		}
 	}
 
-	if err := m.saveMeta(dstSnapshotPath, snapshot); err != nil {
+	if err := m.saveMeta(snapshot); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-// TODO handle className
-func (m *StorageFileSystemModule) RestoreSnapshot(ctx context.Context, className, snapshotID string) error {
-	if err := ctx.Err(); err != nil {
-		return errors.Wrapf(err, "restore snapshot aborted, invalid context")
-	}
-
-	metaPath := m.makeMetaFilePath(snapshotID)
-
-	metaData, err := os.ReadFile(metaPath)
+func (m *StorageFileSystemModule) RestoreSnapshot(ctx context.Context, className, snapshotId string) error {
+	snapshot, err := m.loadSnapShotMeta(ctx, className, snapshotId)
 	if err != nil {
-		return errors.Wrapf(err, "Could not read snapshot meta file %v", metaPath)
-	}
-	var snapshot snapshots.Snapshot
-	if err := json.Unmarshal(metaData, &snapshot); err != nil {
-		return errors.Wrapf(err, "Could not unmarshal snapshot meta file %v", metaPath)
+		return errors.Wrapf(err, "load snapshot meta")
```

Comment:
Ideally, the wrapped error would contain the information about the failed action (load snapshot meta in this case), and the wrap message would refer to the action in which it originates (in this case, restore snapshot).